scipy/special/cephes/bdtr.c

```diff
@@ -148,103 +148,89 @@
 
 #include "mconf.h"
 
-double bdtrc(k, n, p)
-int k, n;
-double p;
+double bdtrc(double k, double n, double p)
```

Comment:
Hm I'm not sure that rounding makes sense for `n`; it seems like if `n != floor(n)` then we should return NaN.

OTOH that's now going to be a back compatibility break, so if we want to change it we will need to go through a deprecation cycle.

Reply:
>OTOH that's now going to be a back compatibility break, so if we want to change it we will need to go through a deprecation cycle.

I'm not sure I follow. Currently, the `bdtr()` throws a runtime warning. Are you suggesting to depracate the current behavior and the `floor()` behavior proposed here for a `double n` argument where 
```
if (n != floor(n)) {
    sf_error("bdtrc", SF_ERROR_DOMAIN, NULL);
    return NPY_NAN;
}
```
If I understand correctly please let me know. I haven't done any depracations in SciPy before but I think I've seen one or two annotations for them. I'm guessing the typical process is add the depracated annotation for `X` release cycles then change the behavior. Please let me know what we'd like to do here, e.g. if my understanding is correct.

Reply:
> Are you suggesting to depracate the current behavior and the floor() behavior proposed here

Yes, that’s what I’m proposing. Before getting into the details of deprecating, however, do we agree that nonintegral `n` doesn’t seem like a reasonable input? We should think through this one carefully.